openwisp_utils/bots/ci_failure/bot.py

```
@@ -0,0 +1,483 @@
+        self.repository_name = os.environ.get("REPOSITORY")
+        self.pr_number = os.environ.get("PR_NUMBER")
+
+        # Initialize with None values if missing - bot will still try to comment
+        self.github = None
+        self.repo = None
+
+        if self.github_token and self.repository_name:
+            try:
+                self.github = Github(self.github_token)
+                self.repo = self.github.get_repo(self.repository_name)
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)